provisioning/device/src/interfaces.ts

```diff
@@ -28,26 +31,36 @@ export interface ProvisioningTransportOptions {
 }
 
 /**
- * Device configuration returned when registration is complete
+ * Device configuration returned when registration using X509 is complete
  */
-export interface ProvisioningDeviceConfiguration {
-  iotHubUri: string;
+export interface X509RegistrationResult {
+  /**
+   * deviceId for the provisioned device
+   */
   deviceId: string;
+  /**
+   * IoT Hub where the provisioned device is located
+   */
+  assignedHub: string;
 }
 
 /**
- * Interface that all provisioning transports must implement in order to support the provisioning service
+ * @private
  */
-
-export interface ProvisioningTransportHandlersBase {
+export interface X509ProvisioningTransport {
   setTransportOptions(options: ProvisioningTransportOptions): void;
   endSession(callback: (err?: Error) => void): void;
+  registerX509(registrationId: string, auth: X509, forceRegistration: boolean, callback: (err?: Error, registrationResult?: X509RegistrationResult, body?: any, result?: any) => void): void;
```

Comment:
>registerX509 [](start = 2, length = 12)

register? no need for the X509 suffix anymore

Reply:
I'd like to be explicit here because a single transport can support x509 and TPM both 

---
In reply to: [153351796](https://github.com/Azure/azure-iot-sdk-node/pull/159#discussion_r153351796) [](ancestors = 153351796)